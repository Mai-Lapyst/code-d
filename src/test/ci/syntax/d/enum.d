/**
 * Selector constants for ucurr_getName().
 */
enum UCurrNameStyle {
    /**
     * Selector for ucurr_getName indicating a symbolic name for a
     * currency, such as "$" for USD.
     * @stable ICU 2.6
     */
    UCURR_SYMBOL_NAME,

    /**
     * Selector for ucurr_getName indicating the long name for a
     * currency, such as "US Dollar" for USD.
     * @stable ICU 2.6
     */
    UCURR_LONG_NAME,

    /**
     * Selector for getName() indicating the narrow currency symbol.
     * The narrow currency symbol is similar to the regular currency
     * symbol, but it always takes the shortest form: for example,
     * "$" instead of "US$" for USD in en-CA.
     *
     * @stable ICU 61
     */
    UCURR_NARROW_SYMBOL_NAME,
}
