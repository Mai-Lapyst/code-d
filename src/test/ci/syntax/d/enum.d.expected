[{"start":0,"end":3,"scope":"comment.block.begin.d"}]
[{"start":0,"end":1,"scope":"source.d"},{"start":1,"end":2,"scope":"keyword.operator.numeric.d"},{"start":2,"end":3,"scope":"source.d"},{"start":3,"end":11,"scope":"variable.d"},{"start":11,"end":12,"scope":"source.d"},{"start":12,"end":21,"scope":"variable.d"},{"start":21,"end":22,"scope":"source.d"},{"start":22,"end":25,"scope":"keyword.control.for.d"},{"start":25,"end":26,"scope":"source.d"},{"start":26,"end":39,"scope":"source.d"},{"start":39,"end":40,"scope":"source.d"},{"start":40,"end":41,"scope":"source.d"},{"start":41,"end":43,"scope":"source.d"}]
[{"start":0,"end":4,"scope":"source.d"}]
[{"start":0,"end":4,"scope":"storage.class.d"},{"start":4,"end":5,"scope":"source.d"},{"start":5,"end":19,"scope":"variable.d"},{"start":19,"end":22,"scope":"source.d"}]
[{"start":0,"end":4,"scope":"source.d"},{"start":4,"end":7,"scope":"comment.block.begin.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":15,"scope":"variable.d"},{"start":15,"end":16,"scope":"source.d"},{"start":16,"end":19,"scope":"keyword.control.for.d"},{"start":19,"end":20,"scope":"source.d"},{"start":20,"end":67,"scope":"source.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":15,"scope":"variable.d"},{"start":15,"end":16,"scope":"keyword.operator.comma.d"},{"start":16,"end":17,"scope":"source.d"},{"start":17,"end":21,"scope":"variable.d"},{"start":21,"end":22,"scope":"source.d"},{"start":22,"end":24,"scope":"variable.d"},{"start":24,"end":25,"scope":"source.d"},{"start":25,"end":26,"scope":"string.double-quoted-string.d"},{"start":26,"end":27,"scope":"string.double-quoted-string.d"},{"start":27,"end":28,"scope":"string.double-quoted-string.d"},{"start":28,"end":29,"scope":"source.d"},{"start":29,"end":32,"scope":"keyword.control.for.d"},{"start":32,"end":33,"scope":"source.d"},{"start":33,"end":38,"scope":"source.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":14,"scope":"entity.name.tag.user-defined-property.d"},{"start":14,"end":15,"scope":"source.d"},{"start":15,"end":18,"scope":"variable.d"},{"start":18,"end":19,"scope":"source.d"},{"start":19,"end":22,"scope":"constant.numeric.float.decimal.d"}]
[{"start":0,"end":8,"scope":"source.d"}]
[{"start":0,"end":4,"scope":"source.d"},{"start":4,"end":21,"scope":"variable.d"},{"start":21,"end":22,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":1,"scope":"source.d"}]
[{"start":0,"end":4,"scope":"source.d"},{"start":4,"end":7,"scope":"comment.block.begin.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":15,"scope":"variable.d"},{"start":15,"end":16,"scope":"source.d"},{"start":16,"end":19,"scope":"keyword.control.for.d"},{"start":19,"end":20,"scope":"source.d"},{"start":20,"end":65,"scope":"source.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":15,"scope":"variable.d"},{"start":15,"end":16,"scope":"keyword.operator.comma.d"},{"start":16,"end":17,"scope":"source.d"},{"start":17,"end":21,"scope":"variable.d"},{"start":21,"end":22,"scope":"source.d"},{"start":22,"end":24,"scope":"variable.d"},{"start":24,"end":25,"scope":"source.d"},{"start":25,"end":26,"scope":"string.double-quoted-string.d"},{"start":26,"end":35,"scope":"string.character.d"},{"start":35,"end":36,"scope":"string.double-quoted-string.d"},{"start":36,"end":37,"scope":"source.d"},{"start":37,"end":40,"scope":"keyword.control.for.d"},{"start":40,"end":41,"scope":"source.d"},{"start":41,"end":46,"scope":"source.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":14,"scope":"entity.name.tag.user-defined-property.d"},{"start":14,"end":15,"scope":"source.d"},{"start":15,"end":18,"scope":"variable.d"},{"start":18,"end":19,"scope":"source.d"},{"start":19,"end":22,"scope":"constant.numeric.float.decimal.d"}]
[{"start":0,"end":8,"scope":"source.d"}]
[{"start":0,"end":4,"scope":"source.d"},{"start":4,"end":19,"scope":"variable.d"},{"start":19,"end":20,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":1,"scope":"source.d"}]
[{"start":0,"end":4,"scope":"source.d"},{"start":4,"end":7,"scope":"comment.block.begin.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":15,"scope":"variable.d"},{"start":15,"end":16,"scope":"source.d"},{"start":16,"end":19,"scope":"keyword.control.for.d"},{"start":19,"end":20,"scope":"source.d"},{"start":20,"end":27,"scope":"source.d"},{"start":27,"end":28,"scope":"source.d"},{"start":28,"end":29,"scope":"source.d"},{"start":29,"end":30,"scope":"source.d"},{"start":30,"end":40,"scope":"variable.d"},{"start":40,"end":41,"scope":"source.d"},{"start":41,"end":44,"scope":"variable.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":51,"scope":"variable.d"},{"start":51,"end":52,"scope":"source.d"},{"start":52,"end":60,"scope":"variable.d"},{"start":60,"end":61,"scope":"source.d"},{"start":61,"end":67,"scope":"variable.d"},{"start":67,"end":69,"scope":"source.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":10,"scope":"variable.d"},{"start":10,"end":11,"scope":"source.d"},{"start":11,"end":17,"scope":"variable.d"},{"start":17,"end":18,"scope":"source.d"},{"start":18,"end":26,"scope":"variable.d"},{"start":26,"end":27,"scope":"source.d"},{"start":27,"end":33,"scope":"variable.d"},{"start":33,"end":34,"scope":"source.d"},{"start":34,"end":36,"scope":"keyword.operator.identity.d"},{"start":36,"end":37,"scope":"source.d"},{"start":37,"end":44,"scope":"variable.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":47,"scope":"variable.d"},{"start":47,"end":48,"scope":"source.d"},{"start":48,"end":51,"scope":"variable.d"},{"start":51,"end":52,"scope":"source.d"},{"start":52,"end":59,"scope":"variable.d"},{"start":59,"end":60,"scope":"source.d"},{"start":60,"end":68,"scope":"variable.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":13,"scope":"variable.d"},{"start":13,"end":14,"scope":"keyword.operator.comma.d"},{"start":14,"end":15,"scope":"source.d"},{"start":15,"end":18,"scope":"variable.d"},{"start":18,"end":19,"scope":"source.d"},{"start":19,"end":21,"scope":"variable.d"},{"start":21,"end":22,"scope":"source.d"},{"start":22,"end":28,"scope":"variable.d"},{"start":28,"end":29,"scope":"source.d"},{"start":29,"end":34,"scope":"variable.d"},{"start":34,"end":35,"scope":"source.d"},{"start":35,"end":38,"scope":"variable.d"},{"start":38,"end":39,"scope":"source.d"},{"start":39,"end":47,"scope":"variable.d"},{"start":47,"end":48,"scope":"source.d"},{"start":48,"end":52,"scope":"variable.d"},{"start":52,"end":54,"scope":"source.d"},{"start":54,"end":57,"scope":"keyword.control.for.d"},{"start":57,"end":58,"scope":"source.d"},{"start":58,"end":67,"scope":"source.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":8,"scope":"string.double-quoted-string.d"},{"start":8,"end":9,"scope":"string.double-quoted-string.d"},{"start":9,"end":10,"scope":"string.double-quoted-string.d"},{"start":10,"end":11,"scope":"source.d"},{"start":11,"end":18,"scope":"variable.d"},{"start":18,"end":19,"scope":"source.d"},{"start":19,"end":21,"scope":"variable.d"},{"start":21,"end":22,"scope":"source.d"},{"start":22,"end":23,"scope":"string.double-quoted-string.d"},{"start":23,"end":25,"scope":"string.character.d"},{"start":25,"end":26,"scope":"string.double-quoted-string.d"},{"start":26,"end":27,"scope":"string.double-quoted-string.d"},{"start":27,"end":28,"scope":"source.d"},{"start":28,"end":31,"scope":"keyword.control.for.d"},{"start":31,"end":32,"scope":"source.d"},{"start":32,"end":46,"scope":"source.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"}]
[{"start":0,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"keyword.operator.numeric.d"},{"start":6,"end":7,"scope":"source.d"},{"start":7,"end":14,"scope":"entity.name.tag.user-defined-property.d"},{"start":14,"end":15,"scope":"source.d"},{"start":15,"end":18,"scope":"variable.d"},{"start":18,"end":19,"scope":"source.d"},{"start":19,"end":21,"scope":"constant.numeric.integer.decimal.d"}]
[{"start":0,"end":8,"scope":"source.d"}]
[{"start":0,"end":4,"scope":"source.d"},{"start":4,"end":28,"scope":"variable.d"},{"start":28,"end":29,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":2,"scope":"source.d"}]
[{"start":0,"end":1,"scope":"source.d"}]